sdk/core/src/test/java/com/klaviyo/core/networking/KlaviyoNetworkMonitorTest.kt

```diff
@@ -167,4 +172,25 @@ internal class KlaviyoNetworkMonitorTest : BaseTest() {
         netCallbackSlot.captured.onAvailable(mockk())
         assertEquals(5, callCount)
     }
+
+    @Test()
+    fun `Concurrent modification exception doesn't get thrown on concurrent observer access`() = runTest {
```

Comment:
Its very cool you were able to repro this, considering how unreliable concurrency issues can be.